testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestHiveCoercion.java

```diff
@@ -816,10 +904,21 @@ private static void alterTableColumnTypes(String tableName)
         onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_smallint tinyint_to_smallint smallint", tableName));
         onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_int tinyint_to_int int", tableName));
         onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_bigint tinyint_to_bigint bigint", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_string tinyint_to_string string", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_bounded_varchar tinyint_to_bounded_varchar varchar(30)", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN tinyint_to_smaller_varchar tinyint_to_smaller_varchar varchar(3)", tableName));
         onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_int smallint_to_int int", tableName));
         onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_bigint smallint_to_bigint bigint", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_string smallint_to_string string", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_bounded_varchar smallint_to_bounded_varchar varchar(30)", tableName));
+        onHive().executeQuery(format("ALTER TABLE %s CHANGE COLUMN smallint_to_smaller_varchar smallint_to_smaller_varchar varchar(5)", tableName));
```

Comment:
When we are coercing to smaller column we need to ensure that the value would be truncated so for a tinyint of `59` - coercing to varchar(1) should give `5` as its output.

Reply:
Done

Reply:
Thanks @Praveen2112 for the review, I modified `IntegerNumberToVarcharCoercer` to support truncation.